### PR TITLE
Add spanish translation of metaprogramming lesson

### PR DESCRIPTION
**Pull request by erickgnavar**

### CODE REVIEW

File: es/lessons/advanced/metaprogramming.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
+---
+version: 1.0.2
+title: Metaprogramación
+---
+
+La Metaprogramación es el proceso de usar código para escribir código.
+En Elixir esto nos da la habilidad de extender el lenguaje para que encaje con nuestras necesidades y poder cambiar el código dinámicamente.
+Vamos a empezar mirando como Elixir está representado por detrás, luego vamos a modificarlo y finalmente podemos usar este conocimiento para extenderlo.
+
+Una advertencia: La Metaprogramación es tramposa y solo debería ser usada cuando sea realmente necesario.
```
[the remark this answers is not in the document]
Comment (erickgnavar) [reply es/lessons/advanced/metaprogramming.md]: Fixed

File: es/lessons/advanced/metaprogramming.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  _ ->
+    "Hi"
+end
+```
+
+Puede que recuerdes que mencionamos que `if` es un macro en Elixir, aquí lo vemos expandido en su forma interna como un enunciado `case`.
+
+### Macros privados
+
+
```
[the remark this answers is not in the document]
Comment (erickgnavar) [reply es/lessons/advanced/metaprogramming.md]: Fixed

**Comment by doomspork** [comment]
@erickgnavar would you like to wait for @milmazz or should I merge given @vbrazo's approval?

**Comment by erickgnavar** [comment]
Hi @doomspork, I applied the recommendations given by @milmazz so I think you can proceed with the merge.

**Comment by doomspork** [comment]
Great @erickgnavar! @vbrazo speaks Spanish so I've added him to the team as well 😁

**Pull request merged**